### DISCUSSION TITLE
Who is InfoEther, LLC in License and what does it mean for usage?

### DISCUSSION THREAD

**Discussion by rsoesemann** [pmd/pmd#3050] [Q&A] [answered]
Salesforce.com is internally evaluating if they can contribute to PMD but found "InfoEther, LLC" as a legal owner of PMD in the [Licence.md](https://github.com/pmd/pmd/blob/master/LICENSE#L3)

> Copyright (c) 2003-2009, InfoEther, LLC
> All rights reserved.
> 

I know that this Company belonged PMD's original author Tom Copeland back then. But...

1. I saw it was acquired by LivingSocial in 2011, and I'm sure their legal teams will ask me about it since there's no google results in the last few years for them.
2. So the question arises: are they contributing to a project owned by LivingSocial, or one by Tom Copeland? 

They are fearing "patent trolls" or other legal problems when contributing to PMD. Is there any chance to explain those parts of the Licence or even change it to make it more compatible to big contributors?

@adangel @jsotuyod

**Comment by jsotuyod** [comment, marked as answer]
@up2go-rsoesemann you are right.

**Tom Copeland** (@tcopeland) worked as a consultant for InfoEther from 2002. During his time there, he worked on a DARPA project codenamed `Ultralog`, as part of this project PMD came to be.
Sources: https://www.visualcv.com/tom http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.113.8018&rep=rep1&type=pdf

InfoEther was acquired by LivingSocial in 2011, LivingSocial was itself acquired by Groupon on 2016
Source: https://www.crunchbase.com/organization/livingsocial

Now, as far as I know, DARPA funded project's intellectual property belongs to the small business / university that performs it (source https://www.quora.com/If-a-research-project-is-funded-by-a-government-grant-who-owns-the-result). That means PMD's ownership actually belonged to InfoEther, who open sourced it very early on (2002). Assuming both acquisitions by LivingSocial and Groupon included all intelectual property (which most likely did), then those rights currently lie with Groupon.

Groupon is an active OpenSource contributor, using the same licenses PMD does, either Apache 2 or 3-clause BSD; as seen: https://github.com/groupon/

That being said, contributing to PMD should be safe.

Reaching out to Groupon for a statement could be possible if they want to be sure.

Interesting fact: Tom Copeland stayed with InfoEther from 2002 until acquisition, then stayed on LivingSocial until acquisition, and then stayed on with Groupon until earlier this year: https://www.linkedin.com/in/tomcopeland1/

**Comment by rsoesemann** [comment]
FYI @capeterson

**Comment by capeterson** [comment]
Looking at the LICENSE and CONTRIBUTING files I can't seem to find any contributor license agreement or other special terms for contributing to PMD, can I confirm that there is no such agreement?

**Comment by jsotuyod** [comment]
@capeterson you are right. I've the idea of creating one, but so far haven't put the time to do so.

If it's a requirement by Salesforce, we will certainly prioritize that task accordingly.

**Comment by capeterson** [comment]
Not sure if it is or not yet. Talking to some people this week and will let you know if it's an issue. Thanks for the quick response.

**Comment by rsoesemann** [comment]
Sorry forgot to close this. Salesforce has accepted the answers here and gave green light for its developers to contribute. A big thing for the Salesforce-related parts of PMD.

**Comment by jsotuyod** [comment]
@rsoesemann that's huge! Thanks to everyone involved, we are eager to deep this collaboration.
